src/EnergyPlus/OutputProcessor.cc

```diff
@@ -4028,358 +3086,50 @@ namespace OutputProcessor {
-        } else if (unitUpper == "W/M2") {
-            return OutputProcessor::Unit::W_m2;
-        } else if (unitUpper == "M") {
-            return OutputProcessor::Unit::m;
-        } else if (unitUpper == "AH") {
-            return OutputProcessor::Unit::Ah;
-        } else if (unitUpper == "A") {
-            return OutputProcessor::Unit::A;
-        } else if (unitUpper == "V") {
-            return OutputProcessor::Unit::V;
-        } else if (unitUpper == "KMOL/S") {
-            return OutputProcessor::Unit::kmol_s;
-        } else if (unitUpper == "REV/MIN") {
-            return OutputProcessor::Unit::rev_min;
-        } else if (unitUpper == "W/M2-K") {
-            return OutputProcessor::Unit::W_m2K;
-        } else if (unitUpper == "J/KG") {
-            return OutputProcessor::Unit::J_kg;
-        } else if (unitUpper == "KG/KG") {
-            return OutputProcessor::Unit::kg_kg;
-        } else if (unitUpper == "%") {
-            return OutputProcessor::Unit::Perc;
-        } else if (unitUpper == "DEG") {
-            return OutputProcessor::Unit::deg;
-        } else if (unitUpper == "S") {
-            return OutputProcessor::Unit::s;
-        } else if (unitUpper == "KG/M3") {
-            return OutputProcessor::Unit::kg_m3;
-        } else if (unitUpper == "KG/M2-S") {
-            return OutputProcessor::Unit::kg_m2s;
-        } else if (unitUpper == "J/KG-K") {
-            return OutputProcessor::Unit::J_kgK;
-        } else if (unitUpper == "L") {
-            return OutputProcessor::Unit::L;
-        } else if (unitUpper == "K/M") {
-            return OutputProcessor::Unit::K_m;
-        } else if (unitUpper == "M2") {
-            return OutputProcessor::Unit::m2;
-        } else if (unitUpper == "W/M2-C") {
-            return OutputProcessor::Unit::W_m2C;
-        } else if (unitUpper == "RAD") {
-            return OutputProcessor::Unit::rad;
-        } else if (unitUpper == "J/M2") {
-            return OutputProcessor::Unit::J_m2;
-        } else if (unitUpper == "CLO") {
-            return OutputProcessor::Unit::clo;
-        } else if (unitUpper == "W/M-K") {
-            return OutputProcessor::Unit::W_mK;
-        } else if (unitUpper == "W/K") {
-            return OutputProcessor::Unit::W_K;
-        } else if (unitUpper == "K/W") {
-            return OutputProcessor::Unit::K_W;
-        } else if (unitUpper == "KGWATER/S") {
-            return OutputProcessor::Unit::kgWater_s;
-        } else {
-            return OutputProcessor::Unit::unknown;
-        }
-    }
+        // This function is here just for unit test purposes
+        DDOutVar *ddVar = state.dataOutputProcessor->ddOutVars[ddNum];
+        Constant::Units units = ddVar->units;
+        return format(" [{}]", units == Constant::Units::customEMS ? ddVar->unitNameCustomEMS : Constant::unitNames[(int)units]);
+    } // unitStringFromDDitem()
 
 } // namespace OutputProcessor
 
-void SetupOutputVariable(EnergyPlusData &state,
```

Comment:
This type of layering is not good. To refactor, you shouldn't add layers on top of APIs you already have. You add them underneath and retarget the existing APIs. Anyways, I didn't want to do either so I just propagated the changes to all uses and deleted this layer.